cpp/src/ice2slice/Main.cpp

```diff
@@ -176,7 +176,7 @@ compile(const vector<string>& argv)
         }
         else
         {
-            UnitPtr p = Unit::createUnit(false);
+            UnitPtr p = Unit::createUnit("icerpc", false);
```

Comment:
How does this work since icerpc is not a mapped language in Parser.cpp?

Reply:
It doesn't. Good catch that I forgot to add it to the list of known languages.

Reply:
I don't think adding 'icerpc' to the list of known language mappings is a good fix. You could make this first parameter std::optional or treat "" as a special value.

Reply:
I removed `icerpc`. Now `ice2slice` just passes the empty string.
And I added an assert to `mappedName` which enforces that the language string is non-empty.
Since `ice2slice` should never be using mapped names anyways.